src/useCases/archiveCompany.ts

```diff
@@ -0,0 +1,44 @@
+import { prisma } from '@/lib/prisma'
+
+interface ArchiveCompanyResponse {
+  success: boolean
+  message?: string
+}
+
+export const archiveCompany = async (
+  companyId: string
+): Promise<ArchiveCompanyResponse> => {
+  const existingCompany = await prisma.company.findUnique({
+    where: {
+      id: companyId,
+    },
+  })
+  if (!existingCompany) {
+    return {
+      success: false,
+      message: 'Company not found',
+    }
+  }
+
+  if (existingCompany.archived_at !== null) {
+    return {
+      success: true,
+      message: 'Company already archived',
+    }
+  }
+
+  // Archive the company
+  await prisma.company.update({
+    where: {
+      id: companyId,
+    },
+    data: {
+      archived_at: new Date(),
+    },
+  })
+
+  return {
+    success: true,
+    message: 'Company archived successfully',
```

Comment:
Message attribute should be used only for general errors.

In this case we need to return `archivedCompany` object like this:
<img width="551" alt="Captura de Tela 2023-05-06 às 2 00 29 PM" src="https://user-images.githubusercontent.com/335095/236617505-3895b5db-025c-46c0-82b0-5dfb37d08658.png">